core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -375,13 +400,22 @@ public void halfClose() {
 
   @Override
   public void sendMessage(ReqT message) {
-    Preconditions.checkState(stream != null, "Not started");
     Preconditions.checkState(!cancelCalled, "call was cancelled");
     Preconditions.checkState(!halfCloseCalled, "call was half-closed");
     try {
       // TODO(notcarl): Find out if messageIs needs to be closed.
       InputStream messageIs = method.streamRequest(message);
-      stream.writeMessage(messageIs);
+      if (!useGet()) {
+        stream.writeMessage(messageIs);
+      } else {
+        // Encode the request message and put it in the request headers.
+        byte[] payload = new byte[messageIs.available()];
```

Comment:
Available is not necessarily right. Only if KnownLength is implemented is it trustworthy.